sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/LogsHelper.cs

```diff
@@ -85,12 +107,23 @@ internal static List<TelemetryItem> OtelToAzureMonitorLogs(Batch<LogRecord> batc
                     }
                     else
                     {
-                        properties.Add(item.Key, item.Value.ToString().Truncate(SchemaConstants.KVP_MaxValueLength) ?? "null");
+                        try
+                        {
+                            string truncatedKey = item.Key.Truncate(SchemaConstants.MessageData_Properties_MaxKeyLength);
+                            if (!properties.ContainsKey(truncatedKey))
+                            {
+                                properties.Add(item.Key, item.Value.ToString().Truncate(SchemaConstants.KVP_MaxValueLength)?.Truncate(SchemaConstants.MessageData_Properties_MaxValueLength)!);
```

Comment:
There are two more `item.Value.ToString()` calls above in this method.

Reply:
Too many exception handler will slow down the performance. This entire thing needs to be redesigned. I will cover two more item.Value.ToString() in this PR. Even if we miss, there is a handler in the caller that will handle the exception and drops the entire telemetry. Better to handle at lower level as it just misses the property values.

Reply:
Fixed 3914045ba27aa9a604518d182b329e7e2cfca236

Reply:
@rajkumar-rangaraj what's the plan to redesign?

Reply:
No plans yet, I rely on you to plan it out.

Reply:
> Too many exception handler will slow down the performance. 

I think just adding try/catch blocks shouldn't bring down performance. It should only be an issue if we repeatedly catch exceptions, right? We don't expect to run into such situations in the common case.